src/meta/meta_split_service.cpp

```diff
@@ -293,5 +293,165 @@ void meta_split_service::on_add_child_on_remote_storage_reply(error_code ec,
+        response.__set_hint_msg(fmt::format("app({}) is not splitting", req.app_name));
+        derror_f("{} split failed, {}", control_type_str(control_type), response.hint_msg);
+        return;
+    }
+
+    if (req.parent_pidx >= 0 && (control_type == split_control_type::PSC_PAUSE ||
+                                 control_type == split_control_type::PSC_RESTART)) {
+        do_control_single(std::move(app), std::move(rpc));
+    } else {
+        do_control_all(std::move(app), std::move(rpc));
+    }
+}
+
+void meta_split_service::do_control_single(std::shared_ptr<app_state> app, control_split_rpc rpc)
+{
+    const auto &req = rpc.request();
+    const std::string &app_name = req.app_name;
+    const int32_t &parent_pidx = req.parent_pidx;
+    const auto &control_type = req.control_type;
+    auto &response = rpc.response();
+
+    if (parent_pidx >= app->partition_count / 2) {
+        response.err = ERR_INVALID_PARAMETERS;
+        response.__set_hint_msg(fmt::format("invalid parent partition index({})", parent_pidx));
+        derror_f("{} split for app({}) failed, {}",
+                 control_type_str(control_type),
+                 app_name,
+                 response.hint_msg);
+        return;
+    }
+
+    auto iter = app->helpers->split_states.status.find(parent_pidx);
+    if (iter == app->helpers->split_states.status.end()) {
+        response.err = control_type == split_control_type::PSC_PAUSE ? ERR_CHILD_REGISTERED
+                                                                     : ERR_INVALID_STATE;
+        response.__set_hint_msg(fmt::format("partition[{}] is not splitting", parent_pidx));
+        derror_f("{} split for app({}) failed, {}",
+                 control_type_str(control_type),
+                 app_name,
+                 response.hint_msg);
+        return;
+    }
+
+    if (control_type == split_control_type::PSC_PAUSE) { // pause
+        if (iter->second == split_status::CANCELING) {
+            response.err = ERR_INVALID_STATE;
+            response.__set_hint_msg(fmt::format("partition[{}] is canceling", parent_pidx));
+            derror_f("{} split for app({}) failed, {}",
+                     control_type_str(control_type),
+                     app_name,
+                     response.hint_msg);
+            return;
+        }
+
+        if (iter->second == split_status::PAUSING || iter->second == split_status::PAUSED) {
+            dwarn_f("duplicated pause request for app({}), partition[{}]", app_name, parent_pidx);
+            response.err = ERR_OK;
+            return;
+        }
+
+        iter->second = split_status::PAUSING;
+        response.err = ERR_OK;
+        ddebug_f("app({}) partition[{}] pausing split succeed", app_name, parent_pidx);
+    } else { // restart
+        if (iter->second == split_status::SPLITTING) {
+            dwarn_f("duplicated restart request for app({}) partition[{}]", app_name, parent_pidx);
+            response.err == ERR_OK;
```

Comment:
same, I think it should return `err` like `err_invalid_stat` but not `ok`

Reply:
And the `log` show is `duplicated restart`, the `SPLITTING` may be not trigered by `restart`